### PR TITLE
Fix chart directory relative path bug

### LINKED ISSUE #516: Client and server use different chart tarball formats
The tarball format generated by `helm chart package` is different from the one assumed by `pkg/chart`. The former generates:

```
drwxr-xr-x  0 247269  5000        0 Mar 29 19:20 .
-rw-r--r--  0 247269  5000      196 Mar 29 21:15 Chart.yaml
drwxr-xr-x  0 247269  5000        0 Mar 29 11:31 docs
drwxr-xr-x  0 247269  5000        0 Mar 29 11:31 hooks
drwxr-xr-x  0 247269  5000        0 Mar 29 21:00 templates
-rw-r--r--  0 247269  5000     6455 Mar 29 21:12 templates/replicatedservice.py
-rw-r--r--  0 247269  5000     2690 Mar 29 15:54 templates/replicatedservice.py.schema
```

but the latter expects:

```
drwxr-xr-x  0 jackgr 5000        0 Mar 29 19:20 replicatedservice/
-rw-r--r--  0 jackgr 5000      196 Mar 29 21:15 replicatedservice/Chart.yaml
drwxr-xr-x  0 jackgr 5000        0 Mar 29 11:31 replicatedservice/docs/
drwxr-xr-x  0 jackgr 5000        0 Mar 29 11:31 replicatedservice/hooks/
drwxr-xr-x  0 jackgr 5000        0 Mar 29 21:00 replicatedservice/templates/
-rw-r--r--  0 jackgr 5000     6455 Mar 29 21:12 replicatedservice/templates/replicatedservice.py
-rw-r--r--  0 jackgr 5000     2690 Mar 29 15:54 replicatedservice/templates/replicatedservice.py.schema
```

@sparkprime did a quick survey of rpm and deb packages and found that there is no convention there for including a prefix, even in the tar.gz inside the deb package. rpm is a cpmioball, deb is an arball containing two tarballs, and none of these things have any prefixes.

That said, a prefix is useful to avoid tarbombing, where files unpacked from an archive overwrite local files. 

For now, we should fix this problem by adding a prefix in the packaging process, rather than by removing the prefix assumption from the unpackaging process, since that's cleaner, although it means departing with precedent.

### PR DESCRIPTION
**Pull request by jackgr**
Closes #516.

/cc @technosophos

### CODE REVIEW

File: pkg/chart/chart.go [hunk 1]
```diff
@@ -220,20 +220,25 @@ func fname(name string) string {
 // If you are just reading the Chart.yaml file, it is substantially more
 // performant to use LoadChartfile.
 func LoadDir(chart string) (*Chart, error) {
-	if fi, err := os.Stat(chart); err != nil {
+	dir, err := filepath.Abs(filepath.Clean(chart))
```
Comment (technosophos) [inline pkg/chart/chart.go]: IIRC, Abs() calls Clean(), so this is redundant.

Comment (jackgr) [inline pkg/chart/chart.go]: Yes, it cleans after checking for absolute. Removed the call to Clean.

**Comment by jackgr** [comment]
@technosophos thanks for the review!

We really should have a test that takes a chart through its various forms. Start with a dir, then save to an archive, then load from the archive and compare to the original. Would have caught this problem.

I'll add an issue for that and get to it later, unless someone else beats me to it.

**Comment by jackgr** [comment]
Created issue #525.

**Comment by jackgr** [comment]
#474

**Pull request merged**